cmd/kraft/kraft.go

```diff
@@ -85,6 +90,92 @@ func New() *cobra.Command {
 	return cmd
 }
 
+func (k *Kraft) PersistentPre(cmd *cobra.Command, args []string) error {
+	if config.G[config.KraftKit](cmd.Context()).NoCheckUpdates {
+		return nil
+	}
```

Comment:
I would put all of this logic into an internal package, e.g.:

```go
import "kraftkit.sh/internal/updates"

func (k *Kraft) PersistentPre(cmd *cobra.Command, args []string) error {
  ctx := cmd.Context()
  if err := update.CheckForUpdates(cmd.Context()); err != nil {
    log.G(ctx).Debugf("could not check for updates: %v", err)
  }
}

Reply:
When you migrate the code over, you can indeed `return err` instead of using the logging.